src/QuestFactory.ts

```diff
@@ -100,7 +101,7 @@ export class QuestFactory {
 			instance.addGoal(new goalType(instance, goal.config, player));
 		}
 
-		instance.on('progress', (progress) => {
+		instance.on('progress', (progress: ISerializedQuestGoal['progress']) => {
```

Comment:
nit:
Can use interface 'IQuestGoalProgress' that is already made in the QuestGoal file. I personally don't like the magic string since it doesn't get changed with 'refactor'.